crates/dojo-world/src/migration/strategy.rs

```diff
@@ -98,16 +98,26 @@ pub fn prepare_for_migration(
 
     // If world needs to be migrated, then we expect the `seed` to be provided.
     if let Some(world) = &mut world {
-        let salt =
-            seed.map(poseidon_hash_single).ok_or(anyhow!("Missing seed for World deployment."))?;
+        let salt = poseidon_hash_single(seed);
 
         world.salt = salt;
-        world.contract_address = get_contract_address(
+        let generated_world_address = get_contract_address(
             salt,
             diff.world.original_class_hash,
-            &[base.as_ref().unwrap().diff.original],
+            &[base.as_ref().unwrap().diff.original_class_hash],
             FieldElement::ZERO,
         );
+
+        if let Some(world_address) = world_address {
+            if world_address != generated_world_address {
+                bail!(
+                    "Calculated world address doesn't match provided world address.\nIf you are \
+                     deploying with custom seed make sure `world_address` is not configured in \
+                     `Scarb.toml`"
```

Comment:
`world_address` can be defined right? It's just a matter of matching values iirc.
```suggestion
                    "Calculated world address doesn't match provided world address.\nIf you are \
                     deploying with custom seed make sure `world_address` is correctly configured (or not set) in \
                     `Scarb.toml`"
```

Reply:
yeah this sounds better, thanks!